packages/mobile/src/i18n/useChangeLanguage.test.tsx

```diff
@@ -0,0 +1,54 @@
+import { renderHook } from '@testing-library/react-hooks'
+import React from 'react'
+import { Provider } from 'react-redux'
+import i18n from 'src/i18n'
+import * as I18nSlice from 'src/i18n/i18nSlice'
+import useChangeLanguage from 'src/i18n/useChangeLanguage'
+import Logger from 'src/utils/Logger'
+import { createMockStore } from 'test/utils'
+import { mocked } from 'ts-jest/utils'
+
+jest.mock('src/i18n')
+
+const mockedI18n = mocked(i18n)
+
+const changeLanguageSpy = jest.spyOn(I18nSlice, 'setLanguage')
+const loggerErrorSpy = jest.spyOn(Logger, 'error')
+
+const renderHookWithProvider = () => {
+  const store = createMockStore()
+  return renderHook(() => useChangeLanguage(), {
+    wrapper: (component) => (
+      <Provider store={store}>{component?.children ? component.children : component}</Provider>
+    ),
+  })
+}
+
+describe('useChangeLanguage', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should handle changing language', async () => {
+    const setLanguageSpy = jest.fn().mockResolvedValue(true)
+    mockedI18n.changeLanguage.mockImplementationOnce(setLanguageSpy)
+
+    const { result } = renderHookWithProvider()
+    await result.current('pt-BR')
+
+    expect(setLanguageSpy).toHaveBeenCalledWith('pt-BR')
+    expect(changeLanguageSpy).toHaveBeenCalledWith('pt-BR')
+  })
```

Comment:
Can we check that the redux action is actually dispatched?

Reply:
it is! i don't know what is wrong with my brain today....but i named these the wrong way around. i've updated this now - the `setLanguageSpy` is watching the setLanguage action on the i18n slice (and `changeLanguageSpy` watching the i18n instance) :)

Reply:
Looks good now.